src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs

```diff
@@ -192,6 +192,25 @@ private int RunScript(ScriptOptions options, SourceText code, ErrorLogger errorL
                 _compiler.ReportErrors(e.Diagnostics, _console.Error, errorLogger);
                 return CommonCompiler.Failed;
             }
+            catch (Exception e)
+            {
+                var exceptionLines = e.ToString().Split(new[] { Environment.NewLine }, StringSplitOptions.None);
+
+                var idxLastSubmissionLine = -1;
+                for (var idx = 0; idx < exceptionLines.Length; ++idx)
+                {
+                    if (exceptionLines[idx].StartsWith("   at Submission#0.", StringComparison.Ordinal))
+                    {
+                        idxLastSubmissionLine = idx;
+                    }
+                }
+
+                var submissionLines = exceptionLines.Take(idxLastSubmissionLine + 1);
+
+                _console.Out.WriteLine(string.Join(Environment.NewLine, submissionLines));
+
+                return 2;
```

Comment:
>2 [](start = 23, length = 1)

Returning e.HResult might be useful as it would allow to distinguish different error types based on the exit code.

Reply:
👍 good idea. Will do.